bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -483,6 +509,7 @@ impl Connection {
 #[derive(Default)]
 struct Context {
     waker: Option<Waker>,
+    pending_callback: Option<Box<dyn AsyncCallback>>,
```

Comment:
Since `AsyncClientHelloCallback` doesnt actually store any state (the state is stored on the config context), you can replace this with a bool.

```suggestion
    pending_ch_callback: bool
```

Reply:
AsyncClientHelloCallback doesn't store any state, but future callbacks will. The AsyncCallback is intended to wrap any state passed to the callback so that it can be reused every time the callback is polled. There just isn't any state passed to the client hello callback.